tests/disaster-recovery/mdr/test_app_failover_and_relocate.py

```diff
@@ -0,0 +1,226 @@
+import logging
+import pytest
+import time
+
+
+from ocs_ci.framework.pytest_customization.marks import tier1
+from ocs_ci.framework import config
+from ocs_ci.ocs.acm.acm import AcmAddClusters
+from ocs_ci.ocs.dr.dr_workload import validate_data_integrity
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.node import wait_for_nodes_status, get_node_objs
+from ocs_ci.ocs.resources.pod import wait_for_pods_to_be_running
+from ocs_ci.helpers.dr_helpers import (
+    enable_fence,
+    enable_unfence,
+    get_fence_state,
+    failover,
+    relocate,
+    set_current_primary_cluster_context,
+    set_current_secondary_cluster_context,
+    get_current_primary_cluster_name,
+    get_current_secondary_cluster_name,
+    wait_for_all_resources_creation,
+    wait_for_all_resources_deletion,
+    gracefully_reboot_ocp_nodes,
+)
+from ocs_ci.helpers.dr_helpers_ui import (
+    failover_relocate_ui,
+    verify_failover_relocate_status_ui,
+)
+from ocs_ci.utility import version
+
+logger = logging.getLogger(__name__)
+
+polarion_id_primary_up = ""
+polarion_id_primary_down = "OCS-4346"
+
+
+@tier1
+class TestApplicationFailoverAndRelocate:
+    """
+    Test Failover and Relocate actions for a busybox application
+    """
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request, dr_workload):
+        """
+        If fenced, unfence the cluster and reboot nodes
+        """
+
+        def finalizer():
+            if self.drcluster_name and get_fence_state(self.drcluster_name) == "Fenced":
+                enable_unfence(self.drcluster_name)
+                gracefully_reboot_ocp_nodes(self.namespace, self.drcluster_name)
+
+        request.addfinalizer(finalizer)
+
+    @pytest.mark.parametrize(
+        argnames=["primary_cluster_down"],
+        argvalues=[
+            pytest.param(
+                False,
+                marks=pytest.mark.polarion_id(polarion_id_primary_up),
+                id="primary_up",
+            ),
+            pytest.param(
+                True,
+                marks=pytest.mark.polarion_id(polarion_id_primary_down),
+                id="primary_down",
+            ),
+        ],
+    )
+    def test_application_failover_and_relocate(
+        self,
+        setup_acm_ui,
+        primary_cluster_down,
+        nodes_multicluster,
+        dr_workload,
+        node_restart_teardown,
+    ):
+        """
+        Tests to verify application failover and relocate between managed clusters
+        There are two test cases:
+            1) Failover and relocate of only application between managed clusters
+            2) Failover to secondary cluster when primary cluster is DOWN and Relocate
+                back to primary cluster once it recovers
+
+        This test is also compatible to be run from UI,
+        pass the yaml conf/ocsci/dr_ui.yaml to trigger it.
```

Comment:
better to have this as part of "usage info"

Reply:
+1

Reply:
Will take in next PR